src/scripts/main.js

```diff
@@ -1,3 +1,274 @@
+  cell.className = 'field-cell';
+  cell.innerText = '';
+};
+
+const getEmptyCells = () => {
+  const emptyCells = [];
+
+  for (let r = 0; r < gameBoard.length; r++) {
+    for (let c = 0; c < gameBoard[r].length; c++) {
+      if (gameBoard[r][c] === 0) {
+        emptyCells.push(
+          {
+            r,
+            c,
+          }
+        );
+      }
+    }
+  }
+
+  return emptyCells;
+};
+
+const checkResult = (value) => {
+  if (value === WINNING_RESULT) {
+    isWon = true;
+  }
+};
+
+const gameOver = () => {
+  document.removeEventListener('keyup', gameControl);
+};
+
+const showWinningMessage = () => {
+  messageWin.classList.remove('hidden');
+};
+
+const showLossMessage = () => {
+  messageLose.classList.remove('hidden');
+};
+
+const updateGame = () => {
+  for (let r = 0; r < ROWS; r++) {
+    for (let c = 0; c < COLUMNS; c++) {
+      const tableCell = gameTable.rows[r].cells[c];
+      const currentValue = gameBoard[r][c];
+
+      resetCell(tableCell);
+
+      if (currentValue !== 0) {
+        tableCell.classList.add(`field-cell--${currentValue}`);
+        tableCell.innerText = `${currentValue}`;
+      }
+    }
+  }
+
+  scoreInfo.innerText = score;
+};
+
+const filterZeros = (row) => {
+  return row.filter(item => item !== 0);
+};
+
+const slide = (row) => {
+  const newRow = filterZeros(row);
+
+  for (let i = 0; i < newRow.length - 1; i++) {
+    if (newRow[i] === newRow[i + 1]) {
+      const newValue = newRow[i] * 2;
+
+      newRow[i] = newValue;
+      newRow[i + 1] = 0;
+
+      score += newRow[i];
+      checkResult(newValue);
+    }
+  }
+
+  const rowResult = filterZeros(newRow);
+
+  while (rowResult.length < COLUMNS) {
+    rowResult.push(0);
+  }
+
+  return rowResult;
+};
+
+const slideLeft = (currentBoard) => {
+  for (let i = 0; i < ROWS; i++) {
+    let row = currentBoard[i];
+
+    row = slide(row);
+    currentBoard[i] = row;
+  }
+};
+
+const slideRight = (currentBoard) => {
+  for (let i = 0; i < ROWS; i++) {
+    let row = currentBoard[i];
+
+    row = slide(row.reverse());
+    currentBoard[i] = row.reverse();
+  }
+};
+
+const slideUp = (currentBoard) => {
+  for (let i = 0; i < ROWS; i++) {
+    const verticalRow = [];
+
+    for (const arr of currentBoard) {
+      verticalRow.push(arr[i]);
+    }
+
+    const row = slide(verticalRow);
+
+    for (let k = 0; k < currentBoard.length; k++) {
+      currentBoard[k][i] = row[k];
+    }
+  }
+};
+
+const slideDown = (currentBoard) => {
+  for (let i = 0; i < ROWS; i++) {
+    const verticalRow = [];
+
+    for (const arr of currentBoard) {
+      verticalRow.push(arr[i]);
+    }
+
+    const row = slide(verticalRow.reverse());
+
+    row.reverse();
+
+    for (let k = 0; k < currentBoard.length; k++) {
+      currentBoard[k][i] = row[k];
+    }
+  }
+};
+
+const getNewCell = () => {
+  const percent = Math.floor(Math.random() * 100) + 1;
+
+  switch (true) {
+    case percent <= 10:
+      return 4;
+
+    default:
+      return 2;
+  }
```

Comment:
I'd avoid using `switch(true)` moreover there could be only one `if`:
```suggestion
    if (percent <= 10) {
      return 4;
    }

    return 2;
```